src/Numeric/Half.hs

```diff
@@ -190,3 +191,6 @@ instance Num Half where
   abs = toHalf . abs . fromHalf
   signum = toHalf . signum . fromHalf
   fromInteger a = toHalf (fromInteger a)
+
+instance Lift Half where
+  lift = return . LitE . FloatPrimL . toRational 
```

Comment:
This creates a value of type `Floating a => a` instead of the more specific `Half`.

This could prevent the code from typechecking, and it'll also break programs compiled with CLaSH which depend on the `Half` constructor being there.